### PR TITLE
[script.module.codequick] 1.0.1+matrix.1

### PR DESCRIPTION
**Pull request by willforde**
### Description
* Added listitem caching
* Changed urlquick to be a wrapper for requests with the same caching system
* Lots of small tweaks

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.module.codequick/addon.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        <import addon="xbmc.python" version="3.0.0"/>
+        <import addon="script.module.youtube.dl" version="19.912.1"/>
+        <import addon="script.module.htmlement" version="1.0.0"/>
+        <import addon="script.module.requests" version="2.22.0"/>
+    </requires>
+    <extension point="xbmc.python.module" library="lib"/>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en_GB">Framework for creating kodi add-on's.</summary>
+        <description lang="en_GB">Codequick is a framework for kodi add-on's. The goal of this framework is to simplify add-on development. This is achieved by reducing the amount of boilerplate code to a minimum, automating tasks like route dispatching and sort method selection. Ultimately allowing the developer to focus primarily on scraping content from websites and passing it to kodi.</description>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline script.module.codequick/addon.xml:13]: GPL-2.0-only or GPL-2.0-or-later

File: script.module.codequick/addon.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        <source>https://github.com/willforde/script.module.codequick</source>
+        <email>willforde+codequick@gmail.com</email>
+        <platform>all</platform>
+        <language>en</language>
```
Comment (enen92) [inline script.module.codequick/addon.xml:17]: remove the language element, this only applies to content

**Review by enen92 — CHANGES_REQUESTED** [review]
Minor and good to go

File: script.module.codequick/addon.xml (hunk 1 above)
Comment (willforde) [reply script.module.codequick/addon.xml:13]: GPL-2.0-only is that ok.

File: script.module.codequick/addon.xml (hunk 2 above)
Comment (willforde) [reply script.module.codequick/addon.xml]: Done

File: script.module.codequick/addon.xml (hunk 1 above)
Comment (enen92) [reply script.module.codequick/addon.xml:13]: only you know, as long as it's an SPDX identifier for the license I'm fine :)

Comment (willforde) [reply script.module.codequick/addon.xml]: I get you now. I was a bit confused. Changed it to GPL-2.0-only

**Review by enen92 — APPROVED** [review]
Thanks

**Pull request merged**